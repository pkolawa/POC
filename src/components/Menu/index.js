import Menu from "./menu";

export { Menu as default };