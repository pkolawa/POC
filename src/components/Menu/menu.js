import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Drawer from "@material-ui/core/Drawer";
import Hidden from "@material-ui/core/Hidden";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import SettingsIcon from "@material-ui/icons/Settings";
import TimelineIcon from "@material-ui/icons/Timeline";
import AutorenewIcon from "@material-ui/icons/Autorenew";
import HomeIcon from "@material-ui/icons/Home";
import { Link } from "react-router-dom";

const drawerWidth = 240;
const useStyles = makeStyles((theme) => ({
  root: {
    flexBasis: drawerWidth,
    "& a": {
      textDecoration: "none",
      display: "inline-flex",
      paddingTop: theme.spacing(1.5),
      paddingBottom: theme.spacing(1.5),
      "& p": {
        display: "inline-block",
        paddingLeft: theme.spacing(1),
      },
    },
  },
  drawerPaper: {
    width: drawerWidth,
    top: "64px",
  },
}));

export default function Menu() {
  const classes = useStyles();

  const drawer = (
    <div>
      <List>
        {["Dashboard", "Configuration", "Monitor", "Runtime"].map(
          (text, index) => (
            <ListItem button key={text}>
              <Link to={`/app/${text.toLowerCase()}`}>
                {text === "Dashboard" && <HomeIcon color="primary" />}
                {text === "Configuration" && <SettingsIcon color="primary" />}
                {text === "Monitor" && <TimelineIcon color="primary" />}
                {text === "Runtime" && <AutorenewIcon color="primary" />}
                <Typography align="center" color="primary">
                  {text}
                </Typography>
              </Link>
            </ListItem>
          )
        )}
      </List>
    </div>
  );

  return (
    <Hidden only={["xs"]}>
      <div className={classes.root}>
        <Drawer
          classes={{
            paper: classes.drawerPaper,
          }}
          variant="permanent"
          open
        >
          {drawer}
        </Drawer>
      </div>
    </Hidden>
  );
}
