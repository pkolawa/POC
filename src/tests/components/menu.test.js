import React from "react";
import { createShallow, getClasses } from "@material-ui/core/test-utils";
import Menu from "../../components/Menu/menu";

let shallow;

beforeAll(() => {
  shallow = createShallow();
});

test("Should Render Menu", () => {
  const wrapper = shallow(<Menu />);
  expect(wrapper).toMatchSnapshot();
});
