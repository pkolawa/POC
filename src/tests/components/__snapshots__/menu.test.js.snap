// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Render Menu 1`] = `
<Hidden
  only={
    Array [
      "xs",
    ]
  }
>
  <div
    className="makeStyles-root-1"
  >
    <WithStyles(ForwardRef(Drawer))
      classes={
        Object {
          "paper": "makeStyles-drawerPaper-2",
        }
      }
      open={true}
      variant="permanent"
    >
      <div>
        <WithStyles(ForwardRef(List))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
            key="Configuration"
          >
            <Link
              to="/configuration"
            >
              <SettingsIcon
                color="primary"
              />
              <WithStyles(ForwardRef(Typography))
                align="center"
                color="primary"
              >
                Configuration
              </WithStyles(ForwardRef(Typography))>
            </Link>
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
            key="Monitor"
          >
            <Link
              to="/monitor"
            >
              <TimelineIcon
                color="primary"
              />
              <WithStyles(ForwardRef(Typography))
                align="center"
                color="primary"
              >
                Monitor
              </WithStyles(ForwardRef(Typography))>
            </Link>
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
            key="Runtime"
          >
            <Link
              to="/runtime"
            >
              <AutorenewIcon
                color="primary"
              />
              <WithStyles(ForwardRef(Typography))
                align="center"
                color="primary"
              >
                Runtime
              </WithStyles(ForwardRef(Typography))>
            </Link>
          </WithStyles(ForwardRef(ListItem))>
        </WithStyles(ForwardRef(List))>
      </div>
    </WithStyles(ForwardRef(Drawer))>
  </div>
</Hidden>
`;
